src/Tooltip/Tooltip.tsx

```diff
@@ -53,68 +112,162 @@ export class Tooltip extends React.Component<TooltipProps, TooltipState> {
                 case "message-container":
                 case "triangle":
                     return;
-                default: this.setState({ toggle: false });
+                default: this.onTooltipToggle(e, false);
             }
         } else {
-            this.setState({ toggle: false });
+            this.onTooltipToggle(null, false);
         }
     }
 
-    /** Forces the tooltip to show */
-    forceShow() {
-        !this.state.toggle && this.setState({ toggle: true });
-    }
-
-    isPositioned(search: string): boolean {
-        const position: string = this.props.position ? this.props.position : "bottom";
-        return position.search(search) === 0;
-    }
-
-    toggleTooltip(state?: boolean, e?: React.MouseEvent<HTMLDivElement>) {
-        if (state !== undefined) {
-            this.setState({ toggle: state });
-        } else {
-            this.setState({ toggle: !this.state.toggle });
-        }
-
-        this.props.onClick && this.props.onClick(e);
+    /**
+     * Forces the tooltip to show
+     */
+    forceShow = (): void => {
+        !this.state.visible && this.onTooltipToggle(null, true);
     }
 
     render() {
         return (
-            <div className={"tooltip-container" + (this.props.className ? ` ${this.props.className}` : "")} id={this.props.id}>
+            <div
+                className={"tooltip-container" + (this.props.className ? ` ${this.props.className}` : "")}
+                id={this.props.id}
+                ref={this.containerRef}
+            >
                 <div
-                    className="icon tooltip-icon"
-                    onClick={(e: React.MouseEvent<HTMLDivElement>) => !this.props.triggerOnHover && this.toggleTooltip(undefined, e)}
-                    onMouseEnter={() => this.props.triggerOnHover && this.toggleTooltip(true)}
-                    onMouseLeave={() => this.props.triggerOnHover && this.toggleTooltip(false)}
+                    className="tooltip-reference"
+                    aria-describedby={this.state.id}
+                    tabIndex={-1}
+                    onClick={(e: React.MouseEvent<HTMLDivElement>) => ((!this.props.trigger && !this.props.triggerOnHover) || this.props.trigger === "click") && this.onTooltipToggle(e)}
+                    onMouseEnter={(e: React.MouseEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, true)}
+                    onMouseLeave={(e: React.MouseEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, false)}
+                    onFocus={(e: React.FocusEvent<HTMLDivElement>) => this.props.trigger === "focus" && this.onTooltipToggle(e)}
+                    onBlur={(e: React.FocusEvent<HTMLDivElement>) => this.onTooltipToggle(e, false)}
```

Comment:
Move all the logic you have for this method `onTooltipToggle` to the method itself. You can know whether you should toggle `true` or `false` based on the `event.type`. Also when you are toggling it off in the `onBlur` method, you should check if it is already toggled off because setting the state will have a lot of effects which you don't want to have. Same goes for `onMouseLeave`

By the way, how is the `focus` and `blur` being triggered here? This is a `div` element with `tabIndex` set to `-1`?

Reply:
by setting tabIndex to -1, it will allow the div to be focusable but not affecting the sequential keyboard navigation, you can refer more on this [link](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex)